cdap-unit-test/src/main/java/co/cask/cdap/test/internal/DefaultStreamManager.java

```diff
@@ -147,6 +157,36 @@ public void send(Map<String, String> headers, ByteBuffer buffer) throws IOExcept
   }
 
   @Override
+  public void send(File file, String contentType) throws Exception {
+    String path = String.format("/v3/namespaces/%s/streams/%s/batch", streamId.getNamespaceId(), streamId.getId());
+    HttpRequest request = new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.POST, path);
+    byte[] fileContent = Files.readAllBytes(Paths.get(file.toURI()));
+    ChannelBuffer content = ChannelBuffers.wrappedBuffer(ByteBuffer.wrap(fileContent));
+    request.setContent(content);
+    request.setHeader(HttpHeaders.Names.CONTENT_LENGTH, content.readableBytes());
+    request.setHeader(HttpHeaders.Names.CONTENT_TYPE, contentType);
+    MockResponder responder = new MockResponder();
+    BodyConsumer bodyConsumer = streamHandler.batch(request, responder, streamId.getNamespaceId(), streamId.getId());
+    Preconditions.checkNotNull(bodyConsumer, "BodyConsumer from stream batch load call should not be null");
+
+    Location location = new LocalLocationFactory().create(file.toURI());
+    try (BufferFileInputStream is = new BufferFileInputStream(location.getInputStream(), 1024 * 1024)) {
+      byte[] chunk = is.read();
```

Comment:
This is bad. It may create a new byte[] on every call. Why need to be so complicated? You can simply use a `FileInputStream` with a `byte[]` or use `ByteStreams.readBytes` from guava.